packages/langium/src/workspace/documents.ts

```diff
@@ -4,18 +4,22 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import type { Diagnostic, Range, TextDocuments } from 'vscode-languageserver';
-import { CancellationToken } from 'vscode-languageserver';
+// re-export 'TextDocument' & friends from 'vscode-languageserver-textdocument' for convenience,
+//  the overhead is very small, just a few kilobytes
+export * from 'vscode-languageserver-textdocument';
```

Comment:
What's the reason for reexporting this dependency here?

Reply:
As soon as one has to customize the document creation/loading stuff clients have to deal the `TextDocument` API.
The import from `vscode-languageserver-textdocument` always felt quite alien to me, especially since `TextDocument` is usually the only thing to be taken from there.
Therefore, I advocate to delegate the issue of from where to import the correct stuff to Langium's API (the foot print is minimal, and `...server-textdocument` provides an ESM/Node16 export).

However, I would be in for wrapping the content into a named export to not pollute Langium's API surface with symbols like `Position`, `Range`, and `TextEdit`.

Reply:
I'm not sure whether reexporting really helps to avoid confusion: it creates one more package from which users can choose to import, and they might wonder whether the definitions are compatible.

In any case, I would avoid reexporting `*` from a foreign package: they might introduce stuff that we don't want to have in our API. Please list the exported symbols explicitly. Is it more than just `TextDocument`?